core/src/main/java/io/seata/core/rpc/netty/NettyClientChannelManager.java

```diff
@@ -209,15 +212,15 @@ private Channel doConnect(String serverAddress) {
     }
     
     private List<String> getAvailServerList(String transactionServiceGroup) throws Exception {
-        List<String> availList = new ArrayList<>();
-        List<InetSocketAddress> availInetSocketAddressList = RegistryFactory.getInstance().lookup(
-            transactionServiceGroup);
-        if (!CollectionUtils.isEmpty(availInetSocketAddressList)) {
-            for (InetSocketAddress address : availInetSocketAddressList) {
-                availList.add(NetUtil.toStringAddress(address));
-            }
+        List<InetSocketAddress> availInetSocketAddressList = RegistryFactory.getInstance()
+                                                                            .lookup(transactionServiceGroup);
+        if (CollectionUtils.isEmpty(availInetSocketAddressList)) {
+            return null;
```

Comment:
please return a emtpy list to avoid NPE.

Reply:
Had modified, the caller method has judgement for null or empty too.